bin/gambdoc.unix.in

```diff
@@ -27,45 +32,55 @@ find_in_path() # exe-name, sets `$exe'
 
 find_browser() # sets `$exe'
 {
-  if [ "@HELP_BROWSER@" != "" ]; then
-    browser_list="@HELP_BROWSER@"
-  else
-    browser_list="lynx firefox mozilla netscape osascript chrome chromium chromium-browser"
-  fi
-
-  browser_list="${GAMBDOC_ARG3_PARAM} $browser_list"
-
+  browser_list="${GAMBDOC_ARG4_PARAM} lynx firefox mozilla netscape osascript chrome chromium chromium-browser"
   for b in $browser_list; do
-    if find_in_path $b; then
+    if find_in_path "$b"; then
       browser=$b
       return 0
     fi
   done
   return 1
 }
 
-operation_help() # sets `$exe'
+open_url_osascript()
 {
-  if find_browser; then
-    url="file://${GAMBITDIR_DOC}/gambit.html#${GAMBDOC_ARG4_PARAM}"
-    case "$browser" in
-      osascript ) $exe <<EOF ;;
+  $exe <<EOF
 tell application "Safari"
     open location "$url"
 end tell
 EOF
-              * ) $exe $url ;;
-    esac
-  else
+}
+
+open_url()
+{
+  url="$1"
+  if ! find_browser; then
     echo "*** WARNING -- none of these browsers can be found to view the documentation:"
     echo "***            $browser_list"
+    exit 2
+  fi
+  case "$browser" in
+    osascript) open_url_osascript "$url";;
+    *) $exe "$url";;
+  esac
+}
+
+operation_help() # sets `$exe'
+{
+  entry="$GAMBDOC_ARG2_PARAM"
+  index="$GAMBDOC_ARG3_PARAM"
+  pattern="$(printf '\t%s\t%s' "$index" "$entry")"
```

Comment:
I don't think `printf` is needed here... just use a literal tab in the string.  That will avoid depending on the existence of `printf`.

Reply:
Literal tabs are very easily nuked by text editors when editing and saving.

Reply:
This will be caught by `git` which will flag the line as having changed.  A nice comment at the end of the line will alert the developer that the code contains tabs that must be preserved.